app/src/main/java/ayds/lisboa/songinfo/home/model/entities/DateFormat.kt

```diff
@@ -0,0 +1,49 @@
+package ayds.lisboa.songinfo.home.model.entities
+
+import ayds.lisboa.songinfo.utils.UtilsInjector
+import ayds.lisboa.songinfo.utils.view.LeapYearCheck
+
+interface DateFormat {
+    fun writeReleaseDatePrecision(song: Song) : String
+}
+
+internal class DateFormatImpl : DateFormat {
+
+    private val leapYearCheck: LeapYearCheck = UtilsInjector.leapYearCheck
```

Comment:
Siempre que se pueda vamos a evitar este tipo de inyección, y hacerlo a través del constructor, para evitar la dependencia de esta clase al injector.
```suggestion
internal class DateFormatImpl(private val leapYearCheck: LeapYearCheck) : DateFormat {
```